app/main.py

```diff
@@ -1,3 +1,70 @@
+from __future__ import annotations
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: Union[int, float]) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union[int, float, Distance]) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(
+                km=self.km + other.km
+            )
+        if type(other) is float or type(other) is int:
```

Comment:
No need to check in other cases
We already know that if it is not Distance it must be int of float
Remove this check!